### PR TITLE
[Core] Callout component!

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

add `Callout` component with props for all the modifiers.

#### Reviewers should focus on:

anything missing?
should we add an example component? (I don't think it's necessary)

### CODE REVIEW

File: packages/core/src/components/callout/callout.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    /** Name of icon to render on left-hand side. */
+    iconName?: IconName;
+
+    /**
+     * String content of optional title element.
+     *
+     * Due to a conflict with the HTML prop types, to provide JSX content simply pass
+     * `<h5>JSX title content<h5>` as first `children` element instead of using this prop.
+     */
+    title?: string;
```
Comment (giladgray) [inline packages/core/src/components/callout/callout.tsx]: see `labelElement` feature, exactly the same thing happened here: #1378

File: packages/core/test/callout/calloutTests.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
```
Comment (giladgray) [inline packages/core/test/callout/calloutTests.tsx]: 😢

File: packages/core/src/components/callout/callout.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+
+import { Classes, IIntentProps, IProps } from "../../common";
+import { IconName } from "../icon/icon";
+
+/** This component also supports the full range of HTML `<div>` props. */
+export interface ICalloutProps extends IIntentProps, IProps {
+    /** Name of icon to render on left-hand side. */
```
Comment (cmslewis) [inline packages/core/src/components/callout/callout.tsx]: nit: We're inconsistent about including articles like `a`, `an`, and `the`. I rather like them.

File: packages/core/src/components/callout/callout.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        const { className, children, iconName, intent, title, ...htmlProps } = this.props;
+        const classes = classNames(
+            Classes.CALLOUT,
+            Classes.intentClass(intent),
+            Classes.iconClass(iconName),
+            className,
+        );
+        return (
+            <div className={classes} {...htmlProps}>
+                {title && <h5>{title}</h5>}
```
Comment (cmslewis) [inline packages/core/src/components/callout/callout.tsx]: FWIW I realize this syntax is functional, but I still find it pretty confusing at a glance.

File: packages/core/src/components/callout/callout.tsx (hunk 3 above)
Comment (giladgray) [reply packages/core/src/components/callout/callout.tsx]: forgotten articles? sounds like FAKE NEWS!

File: packages/core/src/components/callout/callout.tsx (hunk 4 above)
Comment (giladgray) [reply packages/core/src/components/callout/callout.tsx]: not only is it functional, but it appears in React's own documentation

**Comment by cmslewis** [comment]
🚢  🤡

**Pull request merged**